architecture/remote_commands.adoc

```diff
@@ -0,0 +1,36 @@
+= Remote Commands
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+OpenShift takes advantage of a feature built into Kubernetes to support
+executing commands in containers. This is implemented using HTTP along with a
+multiplexed streaming protocol such as link:http://www.chromium.org/spdy[*SPDY*]
+or link:https://http2.github.io/[*HTTP/2*].
+
+Developers can link:../using_openshift/executing_remote_commands.html[use the CLI] to execute remote commands in containers.
+
+== Server Operation
+The Kubelet handles remote execution requests from clients. Upon receiving a
+request, it upgrades the response, evaluates the request headers to determine
+what streams (`stdin`, `stdout`, or `stderr`) to expect to receive, and waits
+for the client to create the streams.
+
+After the Kubelet has received all the streams, it executes the command in the
+container, copying between the streams and the command's stdin, stdout, and
+stderr, as appropriate. When the command terminates, the Kubelet closes the
+upgraded connection, as well as the underlying one.
+
+Architecturally, there are options for running a command in a container. The
+supported implementation currently in OpenShift invokes `nsenter` directly on
+the node host to enter the container's namespaces prior to executing the
+command. However, a custom implementation could include using `docker exec` and
```

Comment:
This sounds like the custom option is 'docker exec' plus a helper container. Those are actually 2 separate options.